dataInterface/items.js

```diff
@@ -9,15 +9,24 @@ module.exports.getAll = () => {
 }
 
 module.exports.getById = (itemId) => {
-  // TODO: complete writing this function
+  const item = module.exports.items.find(item => item.id == itemId);
+  return item;
 }
 
 module.exports.deleteById = async (itemId) => {
-    // TODO: complete writing this function
+    const index = module.exports.items.findIndex(item => item.id == itemId);
+    if(index>=0){
+      const deletedId = module.exports.items.splice(index,1);
+      return deletedId[0];
+    }
 }
 
 module.exports.updateById = async (itemId, newObj) => {
-    // TODO: complete writing this function
+    const index = module.exports.items.findIndex(item => item.id == itemId);
+    if(index>=0){
+     const updatedId = module.exports.items[index].field = newObj.field;
+     return updatedId[0];
+    }
```

Comment:
I love that these are returning the updated or deleted items! In cases where those items aren't found (index < 0), `undefined` is implicitly returned. The code would be more readable if they instead explicitly returned something like `null`. Just an improvement to consider.